tests/android/browser.spec.ts

```diff
@@ -86,6 +86,44 @@ test('should check', async function({ androidDevice }) {
   await context.close();
 });
 
+test('should take page screenshot', async function({ androidDevice }) {
+  test.info().annotations.push({ type: 'issue', description: 'https://github.com/microsoft/playwright/issues/26342' });
+  test.fixme(true, 'Sometimes fails with Protocol error (Page.captureScreenshot): Unable to capture screenshot');
+  test.fixme(true, 'Regular screenshot has an extra pixel border');
+  test.fixme(true, 'Full page screenshot has repeated content');
+
+  const context = await androidDevice.launchBrowser();
+  const [page] = context.pages();
+
+  await page.setContent(`
+    <html lang="en">
+      <head>
+        <meta name="viewport" content="width=device-width, initial-scale=1.0">
+      </head>
+      <body>
+        <style>
+          * { margin: 0; padding: 0; }
+          div {
+            width: 200px;
+            height: 250px;
+          }
+        </style>
+        <div style="background: red"></div>
+        <div style="background: green"></div>
+        <div style="background: red"></div>
+        <div style="background: green"></div>
+        <div style="background: red"></div>
+        <div style="background: green"></div>
+      </body>
+    </html>
+  `);
+  const screenshot = await page.screenshot({ fullPage: false, scale: 'css' });
+  const fullPageScreenshot = await page.screenshot({ fullPage: true, scale: 'css' });
+  expect(screenshot).toMatchSnapshot('page-screenshot.png');
+  expect(fullPageScreenshot).toMatchSnapshot('fullpage-screenshot.png');
+  await context.close();
```

Comment:
try finally context.close since if toMatchSnapshot fails the context will never close?